onnxruntime/core/mlas/lib/pooling.cpp

```diff
@@ -1342,12 +1330,21 @@ Return Value:
 
 #pragma omp parallel for
     for (int64_t c = 0; c < int64_t(TotalChannelCount); c++) {
-        PoolKernelRoutine(&WorkBlock, 1, Input + c * InputSize, Output + c * OutputSize);
+      PoolKernelRoutine(&WorkBlock, 1, Input + c * InputSize, Output + c * OutputSize);
     }
 
 #else
 
     PoolKernelRoutine(&WorkBlock, TotalChannelCount, Input, Output);
 
 #endif
+#else
+    //
+    // Use an external thread pool if one is provided.
+    // TODO: change to use MlasExecuteThreaded
+    onnxruntime::concurrency::ThreadPool::TryBatchParallelFor(ThreadPool, static_cast<int32_t>(TotalChannelCount), [&](int32_t c) {
+      PoolKernelRoutine(&WorkBlock, 1, Input + c * InputSize, Output + c * OutputSize);
```

Comment:
>Input + c * InputSize, Output + c * OutputSize); [](start = 39, length = 48)

Please, supply elaborate comments on these.

Reply:
I just reorganized the original code. Indeed I don't fully understand it.